rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -301,6 +301,11 @@ std::shared_ptr<rosbag2_storage::SerializedBagMessage> SqliteStorage::read_next(
   bag_message->time_stamp = std::get<1>(*current_message_row_);
   bag_message->topic_name = std::get<2>(*current_message_row_);
 
+  // set start time to current time
+  // and set seek_row_id to the new row id up
+  seek_time_ = bag_message->time_stamp;
+  seek_row_id_ = std::get<3>(*current_message_row_) + 1;
```

Comment:
you're right if you assume row_id is in time order, but it might not be. row_id is used as a tie breaker to preserve message pointer at the same timestamp, otherwise, the timestamp is used for that purpose.

Reply:
@lihui815 But how `seek_row_id_` will help to preserve message pointer at the same timestamp when `row_id` is not in timestamp order? 
I don't see a logic for this in your implementation. Could you please explain, clarify?
 
And again the case with situation when `row_id` stored in out of time stamp order should't be considered in this PR at first place.

Reply:
OK - the behavior is changed to maintain previous behavior. Now it moves to the first instance of the time in question, but does not try to maintain that time when loading new files.

We can try to address out-of-order files separately from this feature.

Reply:
@lihui815 Thanks for trying to introduce clarity in your logic with updates in the description.

Although I am pretty mach assure that your assumption is wrong. You are constructing extra level of ordering using  manual indexing via `seek_row_id_` to handle situation when written two or more messages with the same time stamp. i.e.
> write order
  row0 t1 msgA
  row1 t0 msgB
  row2 t0 msgC

But this situation is not possible in regular scenario. Since we have only one common timer and it's accuracy up to the  nanosecond. Even if two cores will ask timestamp at the same processors tact they will get different timestamp from the timer. 
The rule could be violated if someone will **hack** rosbag and insert record with the same timestamp without really taking it from timer. But in this case it will be **ill formed** rosbag and no guarantee that `row_id` will be unique as well (e.g. it could be hacked the similar way). 
IMO. any way we should not make workarounds inside our readers for ill formed bag files.